Dev docs updates: Internationalization plugin (#224)
strapi/documentation: docs/developer-docs/latest/development/plugins/i18n.md

---
title: Internationalization (i18n) - Strapi Developers Documentation
description: Instructions on how to use Strapi Content API with the Internationalization (i18n) optional plugin
---

# Internationalization (i18n)

> <!-- ? wondering if it would be a good idea to add the 🌍 icon to the h1 — if yes, we should do this for the h1 of other plugins as well -->

Reviewer: Yeah if go that route, reach plugin needs to have an emoji

Reviewer 2: Do we agree that all doc pages will contain this suffix then: `- Strapi (Developers|User) Documentation`?
So another page with short title tag would be like: `GraphQL - Strapi Developers Documentation` in order to fix short title tags?

Author: I think the proper wording would be "Strapi Developer Documentation" and "Strapi User Guide" — I'd like @meganelacheny to confirm on her side :-)

I added this to prevent confusion, in case 2 entries share the same name in both user guide and dev docs. If you think this is unnecessary, please let me know ;-)

Reviewer 2: LGTM

> Internationalization is the design and development of a product, application or document content that enables easy localization for target audiences that vary in culture, region, or language.
>
> -- <cite>[W3C](https://www.w3.org/International/questions/qa-i18n.en#i18n)</cite>

> <!-- ? not sure if using a citation is a good idea to introduce/define i18n — feedback is welcome! -->

Reviewer: Fine by mean so we can properly describe what a locale is

> The Internationalization (i18n) plugin allows Strapi users to create, manage and distribute content in several languages.

Reviewer: For this, we should probably mirror the npm/yarn commands to install as well.

Reviewer 3: +1

It's an optional plugin that you can [install from the Marketplace](/user-docs/latest/plugins/installing-plugins-via-marketplace.md).
<!-- TODO: add programmatic instructions to install plugin? like strapi install i18n-->

The i18n plugin:

- allows admin panel users to create several localized versions of their content (see [user guide](/user-docs/latest/content-manager/translating-content.md))
- allows developers to build localized projects by fetching and consuming the right content depending on the country/language of the audience.

::: tip NOTE
The i18n plugin does not automatically translate the users' content nor adapt the admin interface to languages specificities (like displaying the admin panel in Right To Left (RTL) format).
:::

The i18n plugin adds new features to the [Content API](/developer-docs/latest/developer-resources/content-api/content-api.md):

> - a new [`locale`](#getting-localized-entries-with-the-locale-parameter) parameter to fetch content only for a specified locale

Reviewer 3: Same as above?
```suggestion
- a new [`_locale`](#getting-localized-entries-with-the-locale-parameter) parameter to fetch content only for a specified locale
```

- the ability to create a localized entry, either [from scratch](#creating-a-new-localized-entry) or [for an existing localized entry](#creating-a-localization-for-an-existing-entry)

> ## Getting localized entries with the `locale` parameter

Reviewer: Maybe we should update the content-api doc or include a table for it here also :thinking: 

http://localhost:8080/documentation/developer-docs/latest/developer-resources/content-api/content-api.html#api-endpoints

Author: Thanks Derrick. This is something I already discussed with @Convly 
My first draft actually modified the Content API docs ;-)
But we would like to avoid disseminating "plugins-only" information into the core Content API docs. The idea is to mimic the product behavior: if we remove a plugin, it should not impact the "core" (docs).

However, I could probably simply mention in the core Content API doc that i18n adds some features and link to the /plugins/i18n.md file from there. Will suggest something like this soon ;-)

Author: Suggestion:
See [this commit](https://github.com/strapi/documentation/pull/224/commits/274c2ba6899f1859c3bc578bf8f9de721943dca0)

Reviewer 3: I'm not really a big fan of citing an optional plugin directly from the generic/core documentation, I just feel it adds some complexity & overhead. But maybe stating the params "if i18n is activated" is a good compromise :man_shrugging:

> The `locale` [API parameter](/developer-docs/latest/developer-resources/content-api/content-api.md#api-parameters) can be used to fetch entries only for a specified locale. It takes a locale code as value.

Reviewer 3: ```suggestion
The `_locale` [API parameter](/developer-docs/latest/developer-resources/content-api/content-api.md#api-parameters) can be used to fetch entries only for a specified locale. It takes a locale code as value.
```

:::tip TIP
 To fetch content for a locale, make sure it has been already [added to Strapi in the admin panel GUI](/user-docs/latest/settings/managing-global-settings.md#configuring-internationalization-locales).
:::

The format for a GET request is the following:

```js
GET /{content-type}?_locale={locale-code}
```

If the `locale` parameter isn't defined, it will be set to the default locale. `en` is the default locale when i18n plugin is installed, so by default a GET request to `http://localhost:1337/restaurants` will return the same response as a request to `http://localhost:1337/restaurants?_locale=en`.

::: tip TIP
Another locale can be [set as the default locale](/user-docs/latest/settings/managing-global-settings.md#adding-a-new-locale) in the admin panel.
:::

> <!-- ? Should we be more explicit about what a locale code is? What are the available locales? Is there a list? -->

Reviewer: The list should be in the code base

https://github.com/strapi/strapi/blob/features/i18n/packages/strapi-plugin-i18n/constants/iso-locales.json

<!-- TODO maybe add a link to user guide here if Mégane describes the list of available locales -->

When the i18n plugin is installed, the response to requests includes fields that are specific to internationalization:

- `locale` (string) is the locale code for the content entry
- `localizations` (array) lists the existing localizations for this content entry; these localizations objects have 3 properties:

>   - `id` (number) is the id of the localized content entry

Reviewer 3: It can be a string w/ Mongo so I would say `number|string` or `ID`

    "updated_at": "2021-04-07T13:22:46.617Z",
    "closing_period": null,
    "services": [],
    "opening_times": [],
    "dz": [],
    "cover": null,
    "images": [],
    "categories": [],
    "localizations": []
}
```

:::
::::

## Creating a localization for an existing entry

To create another localization for an existing localized entry, send a POST request to the appropriate URL depending on the type of content:

| Content-Type    | Request URL format                       |
|:----------------|:-----------------------------------------|
| Single Type     | `POST /{content-type}/:id/localizations` |  
| Collection Type | `POST /{content-type}/localizations`     |

::: warning CAUTION
When creating a localization for existing localized entries, the body of the POST request can only accept localized fields.
:::

When sending the request to a single type, Strapi will:

1. use the `id` as base entry for the non-localized fields and copy them in the new entry
2. then create a new entry for the given locale and link it with the base entry.
:::: tabs

::: tab Request

**Example request**

```js
POST http://localhost:1337/restaurants/8/localizations
```

**body**

> <!-- * We never used "body" before, see existing entries in Content API. I think it would be a good idea — so it's even more explicit that this code block is the body of the request. If we agree on this, I should update all existing examples in Content API to follow this format. -->

Reviewer 3: Fine by me, but maybe you can do it in another PR so we don't make this one bigger than it needs to be?

Author: Sure!

  "name": "Restaurant F",
  "description": "Restaurant facing the sea"
}
```

This query:

1. creates a new entry in `en`
2. links the created entry with `restaurant:8` (they will share the same `localizations` object)
3. copies every non-localized fields from `restaurant:8` into the new entry and keeps the localized fields from the request's body

:::

::: tab Response

**Example reponse**

```json
{
    "id": 9,
    "name": "Restaurant F",
    "slug": null,
    "price_range": null,
    "contact_email": null,
    "stars": 1,
    "averagePrice": null,
    "address": null,
    "timestamp": null,
    "short_description": null,
    "since": null,
    "description": "Restaurant facing the sea",
    "menu": null,
    "locale": "en",
    "published_at": "2021-04-08T13:29:53.295Z",
    "created_at": "2021-04-08T13:29:53.297Z",
    "updated_at": "2021-04-08T13:29:53.297Z",
    "closing_period": null,
    "services": [],
    "opening_times": [],
    "dz": [],
    "cover": null,
    "images": [],
    "categories": [],
    "localizations": [
        {
            "id": 8,
            "locale": "fr",
            "published_at": "2021-04-07T13:22:46.589Z"
        }
    ]
}
```

:::

::::

> <!-- ? should we also add an example of creating a localization for a collection type? -->

Reviewer 3: Do you mean for a single type? (since you've already put an example for a collection type)

Reviewer 3: But yeah an example for a single type could be a good thing

Author: @Convly 🤦 well yes. I meant single type — indeed we already POST to the `restaurants` collection type 😅

Author: ✅ Added example [with this commit](https://github.com/strapi/documentation/pull/224/commits/380a98b55dedf8ad1df4ea27da94ca524ab5f561)

---
title: Internationalization (i18n) - Strapi Developer Documentation
description: Instructions on how to use Strapi Content API with the Internationalization (i18n) optional plugin
---

# Internationalization (i18n)
<!-- ? wondering if it would be a good idea to add the 🌍 icon to the h1 — if yes, we should do this for the h1 of other plugins as well -->

> Internationalization is the design and development of a product, application or document content that enables easy localization for target audiences that vary in culture, region, or language.
>

> > -- <cite>[W3C](https://www.w3.org/International/questions/qa-i18n.en#i18n)</cite>

Reviewer 4: I'm not very keen on that idea of using a quote 🤔  I think it might be interesting to link a full definition of i18n in our documentation, but in my opinion something like "For more information about the concept of internationalization, please refer to [...]" would be enough.

<!-- ? not sure if using a citation is a good idea to introduce/define i18n — feedback is welcome! -->

> The Internationalization (i18n) plugin allows Strapi users to create, manage and distribute content in several languages.

Reviewer 4: Actually, instead of having the quote at the beginning, it could be nice to have more info here on internationalization, and especially explain what is a "locale", because you've used "language" but then switched to "locale" and it may be confusing.

The i18n plugin:

- allows admin panel users to create several localized versions of their content (see [user guide](/user-docs/latest/content-manager/translating-content.md))
- allows developers to build localized projects by fetching and consuming the right content depending on the country/language of the audience.

::: tip NOTE

> The i18n plugin does not automatically translate the users' content nor adapt the admin interface to languages specificities (like displaying the admin panel in Right To Left (RTL) format).

Reviewer 4: ```suggestion
The i18n plugin does not automatically translate the users' content nor adapt the admin interface to languages specificities (e.g. displaying the admin panel in Right To Left format).
```

:::

## Installation

> The i18n plugin requires a Strapi application running on version 3.6.0 or higher.

Author: I thought doing so, but seems like the only prerequisite is version 3.6. 
I agree for the sake of consistency — but is it worth it having a dedicated section with just one prerequisite? cc @meganelacheny

> The i18n plugin requires a Strapi application running on version 3.6.0 or higher.

Reviewer 4: Well I think it's time to officially decide how we should display this kind of information, because clearly we may have other docs with other prerequisites in the future, and right now you only have 2 docs to update so it's quite easy. Maybe having a "Prerequisites" section isn't the best for the dev docs because we already have lots of titles and long docs. Perhaps you could create a new type of note with "Prerequisites" as title? What do you think @pwizla ?

> The i18n plugin requires a Strapi application running on version 3.6.0 or higher.

Author: I agree @meganelacheny . By "note" do you mean a custom container (that we sometimes call "callout") [like this](https://vuepress.vuejs.org/guide/markdown.html#default-title)?

> The i18n plugin requires a Strapi application running on version 3.6.0 or higher.

Author: Suggestion in [this commit](https://github.com/strapi/documentation/pull/224/commits/580ee2e9cc5bd85a8bc90488326865cb4c0d6731): 
![Screenshot 2021-04-14 at 19 12 56](https://user-images.githubusercontent.com/4233866/114751474-7a071e80-9d55-11eb-8097-c2f11d95e3b1.png)

::: tip TIP
Another locale can be [set as the default locale](/user-docs/latest/settings/managing-global-settings.md#adding-a-new-locale) in the admin panel.
:::

<!-- ? Should we be more explicit about what a locale code is? What are the available locales? Is there a list? -->
<!-- TODO maybe add a link to user guide here if Mégane describes the list of available locales -->

When the i18n plugin is installed, the response to requests includes fields that are specific to internationalization:

- `locale` (string) is the locale code for the content entry
- `localizations` (array) lists the existing localizations for this content entry; these localizations objects have 3 properties:
  - `id` (number|string) is the id of the localized content entry
  - `locale`(string) is the locale code for the localized content entry
  - `published_at` (string) is the date and time when the localized content entry was [published](/developer-docs/latest/concepts/draft-and-publish.md#publishing-a-draft), in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format

:::: tabs

::: tab Request

**Example request**

```js
GET http://localhost:1337/restaurants?_locale=fr
```

:::

::: tab Response

> **Example response**

Reviewer 5: I think there is too many fields display.
To make it easier to understand, I would drastically simplify JSON objects by just keeping 2/3 fields that are interesting. And I would use some restaurant names that come from FoodAdvisor instead of 
```
"name": "Restaurant A",
"slug": "restaurant-a-1",
```

Author: Thanks Jim!
I wanted to use real-world examples and I copied/pasted the full response that I had with testing the feature.

But you're right, I'll simplify this and update it with FoodAdvisor-based examples.

    "opening_times": [],
    "dz": [],
    "cover": null,
    "images": [],
    "categories": [],
    "localizations": []
  }
]
```

In the example response above:

- restaurant with `"id": 4` is a French (`"locale": "fr"`) localization of the existing restaurant with `"id": 3` (created for the default `en` locale), as shown in the `localizations` object included in the response (see [creating a localization for an existing entry](#creating-a-localization-for-an-existing-entry)).
- restaurant with `"id": "8"` was created from scratch using the API, passing the `locale: fr` in the request body (see [creating a new localized entry](#creating-a-new-localized-entry)).
:::
::::

### Creating a new localized entry

To create a localized entry from scratch, send a POST request to the Content API.

If no locale has been passed in the request body, the entry is created using the default locale for the application:

:::: tabs
::: tab Request

**Example request**

```js
POST http://localhost:1337/restaurants
```

```json
{
  "name": "Restaurant B",
  "slug": "restaurant-b",
  "stars": 1,
  "description": "Restaurant with terrace in front of the lake"
}
```

:::

::: tab Response

> **Example response**

Reviewer 5: Same as the previous comment for the simplification of the JSON object and use some FoodAdvisor data.

```json
{
    "id": 5,
    "name": "Restaurant B",
    "slug": "restaurant-b",
    "price_range": null,
    "contact_email": null,
    "stars": 1,
    "averagePrice": null,
    "address": null,
    "timestamp": null,
    "short_description": null,
    "since": null,
    "description": "Restaurant with terraza in front of the lake",
    "menu": null,
    "locale": "en",
    "published_at": "2021-04-07T12:51:43.681Z",
    "created_at": "2021-04-07T12:51:43.695Z",
    "updated_at": "2021-04-07T12:51:43.695Z",
    "closing_period": null,
    "services": [],
    "opening_times": [],
    "dz": [],
    "cover": null,
    "images": [],
    "categories": [],
    "localizations": []
}
```

:::
::::

To create a localized entry for a locale different from the default one, add the `locale` attribute to the body of the POST request:

:::: tabs
::: tab Request

**Example request**

```js
POST http://localhost:1337/restaurants
```

```json
{
  "name": "Restaurant F",
  "slug": "restaurant-f",
  "stars": 1,
  "description": "Restaurant face à la mer",
  "locale": "fr"
}
```

:::

::: tab Response

> **Example response**

Reviewer 5: Same as the previous comment for the simplification of the JSON object and use some FoodAdvisor data.

```json
{
    "id": 8,
    "name": "Restaurant F",
    "slug": "restaurant-f",
    "price_range": null,
    "contact_email": null,
    "stars": 1,
    "averagePrice": null,
    "address": null,
    "timestamp": null,
    "short_description": null,
    "since": null,
    "description": "Restaurant face à la mer",
    "menu": null,
    "locale": "fr",
    "published_at": "2021-04-07T13:22:46.589Z",
    "created_at": "2021-04-07T13:22:46.617Z",
    "updated_at": "2021-04-07T13:22:46.617Z",
    "closing_period": null,
    "services": [],
    "opening_times": [],
    "dz": [],
    "cover": null,
    "images": [],
    "categories": [],
    "localizations": []
}
```

:::
::::

### Creating a localization for an existing entry

To create another localization for an existing localized entry, send a POST request to the appropriate URL depending on the type of content:

> | Content-Type    | Request URL format                       |

Reviewer 5: Holla todos, big question that needs to be clarified in a Guideline document.
Here I can ping @meganelacheny and @maevalienard cause it will/can impact the wording in the admin interface.
- [ ] Content Type
- [ ] Content-Type
- [ ] Content type
- [ ] Content-type
- [ ] content type
- [ ] content-type

Author: Already [raised the issue](https://github.com/strapi/documentation/pull/199#issuecomment-802971394) with Mégane's "Content-Type Builder" docs.

To me, Content Type without a dash would be the most appropriate.

Reviewer 4: We already had this discussion @lauriejim when I started the user guide and I thought we all agreed on "content-type"? If I'm not mistaken, it's also the way it's written in the company's glossary on Notion  🤔  It can be changed if eventually that's not the best option though.
By the way, as we're discussing wording: in the user guide we also normally write "single type" and "collection type" (in this doc I've seen it written differently, like "Single Type" or "Collection Type" even in the middle of a sentence which looks a bit odd in my opinion).

Author: Indeed, I title-cased Single Type and Collection Type because this is how it's written in most of the dev docs I've seen so far. But maybe it's time to add consistency as well! haha
So I figured out we would title-case concepts that are unique to Strapi 🤷

|:----------------|:-----------------------------------------|

> | Single Type     | `POST /{content-type}/:id/localizations` |  

Reviewer 5: I feel like there is something weird there.
For a Single Type, we presume that there is one entry.
I don't really understand why we need to precise the `:id` for Single Type.
Isn't it for Collection Type instead? Lines are propably reverted.

Author: Indeed, something wrong over there…
Need to check with @Convly

Reviewer: They currently had some bugs with this and are fixing them I think.

| Collection Type | `POST /{content-type}/localizations`     |

::: warning CAUTION
When creating a localization for existing localized entries, the body of the POST request can only accept localized fields.
:::

When sending the request to a collection type, Strapi will:

1. use the `id` as a base entry for the non-localized fields and copy them in the new entry
2. then create a new entry for the given locale and link it with the base entry.
:::: tabs

::: tab Request

**Example request**

```js
POST http://localhost:1337/restaurants/8/localizations
```

```json
{
  "locale": "en",
  "name": "Restaurant F",
  "description": "Restaurant facing the sea"
}
```

This query:

1. creates a new entry in `en`
2. links the created entry with `restaurant:8` (they will share the same `localizations` object)
3. copies every non-localized fields from `restaurant:8` into the new entry and keeps the localized fields from the request's body

:::

::: tab Response

> **Example reponse**

Reviewer 5: Same here for the simplification of the response object and the usage of FA data.

```json
{
    "id": 9,
    "name": "Restaurant F",
    "slug": null,
    "price_range": null,
    "contact_email": null,
    "stars": 1,
    "averagePrice": null,
    "address": null,
    "timestamp": null,
    "short_description": null,
    "since": null,
    "description": "Restaurant facing the sea",
    "menu": null,
    "locale": "en",
    "published_at": "2021-04-08T13:29:53.295Z",
    "created_at": "2021-04-08T13:29:53.297Z",
    "updated_at": "2021-04-08T13:29:53.297Z",
    "closing_period": null,
    "services": [],
    "opening_times": [],
    "dz": [],
    "cover": null,
    "images": [],
    "categories": [],
    "localizations": [
        {
            "id": 8,
            "locale": "fr",
            "published_at": "2021-04-07T13:22:46.589Z"
        }
    ]
}
```

:::

::::

<!-- ? should we also add an example of creating a localization for a collection type? -->

::: tip TIP
The Content-Type should have the [`createlocalization` permission]() enabled, otherwise the POST request will return a `403: Forbidden` status.
<!-- TODO: insert cross-link to user guide on how to activate `createlocalization` permission in the admin panel -->
:::

### Updating an entry

> Currently, it is not possible to change the locale of an existing localized entry using the Content API.

Reviewer 5: I don't think it's possible to do it either from the Content Manager, I don't think it's worth it to mention ` using the Content API.`

npm run strapi install i18n
```

:::

::::

## Usage with Strapi Content API

The i18n plugin adds new features to the [Content API](/developer-docs/latest/developer-resources/content-api/content-api.md):

- a new [`locale`](#getting-localized-entries-with-the-locale-parameter) parameter to fetch content only for a specified locale
- the ability to create a localized entry, either [from scratch](#creating-a-new-localized-entry) or [for an existing localized entry](#creating-a-localization-for-an-existing-entry)

### Getting localized entries with the `locale` parameter

The `locale` [API parameter](/developer-docs/latest/developer-resources/content-api/content-api.md#api-parameters) can be used to fetch entries only for a specified locale. It takes a locale code as value (see [full list of available locales](https://github.com/strapi/strapi/blob/features/i18n/packages/strapi-plugin-i18n/constants/iso-locales.json)).

:::tip TIP
 To fetch content for a locale, make sure it has been already [added to Strapi in the admin panel](/user-docs/latest/settings/managing-global-settings.md#configuring-internationalization-locales).
:::

The format for a GET request is the following:

> ```js

Reviewer: I don't think js would be the right "language" but I'm also not familiar with any language tag in vuepress for these logical statements

If the `locale` parameter isn't defined, it will be set to the default locale. `en` is the default locale when i18n plugin is installed, so by default a GET request to `http://localhost:1337/restaurants` will return the same response as a request to `http://localhost:1337/restaurants?_locale=en`.
.

::: tip TIP
Another locale can be [set as the default locale](/user-docs/latest/settings/managing-global-settings.md#adding-a-new-locale) in the admin panel.
:::

When the i18n plugin is installed, the response to requests includes fields that are specific to internationalization:

- `locale` (string) is the locale code for the content entry
- `localizations` (array) lists the existing localizations for this content entry; these localizations objects have 3 properties:
  - `id` (number|string) is the id of the localized content entry
  - `locale`(string) is the locale code for the localized content entry
  - `published_at` (string) is the date and time when the localized content entry was [published](/developer-docs/latest/concepts/draft-and-publish.md#publishing-a-draft), in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format

:::: tabs

::: tab Request

**Example request**

> ```js

Reviewer: same js isn't the right syntax highlight code :thinking:

    "locale": "fr",
    "localizations": [
      {
        "id": 3,
        "locale": "en",
        "published_at": "2021-04-07T10:10:31.949Z"
      }
    ]
  },
  {
    "id": 8,
    "name": "She's Cake",
    "description": "She's Cake restaurant description in French",
    "locale": "fr",
    "localizations": []
  }
]
```

In the example response above:

- restaurant with `"id": 4` is a French (`"locale": "fr"`) localization of the existing restaurant with `"id": 3` (created for the default `en` locale), as shown in the `localizations` object included in the response (see [creating a localization for an existing entry](#creating-a-localization-for-an-existing-entry)).
- restaurant with `"id": "8"` was created from scratch using the API, passing the `locale: fr` in the request body (see [creating a new localized entry](#creating-a-new-localized-entry)).
:::
::::

### Creating a new localized entry

To create a localized entry from scratch, send a POST request to the Content API.

If no locale has been passed in the request body, the entry is created using the default locale for the application:

:::: tabs
::: tab Request

**Example request**

> ```js

Reviewer: Vuepress uses prismjs and this is their list of supported codes: https://prismjs.com/#supported-languages

Author: Thanks for the tip, Derrick! Honestly I was not too sure about what to use there when I wrote it, and re-used what we did for Content API.
Looking at the list you shared, `http` seems more appropriate.

I changed it [here](https://github.com/strapi/documentation/pull/224/commits/c2a40cb13854a920d4e58ff08e46aa931b36e5e5) and I'll create another PR for other similar code blocks in the repo.

    "localizations": []
}
```

:::
::::

To create a localized entry for a locale different from the default one, add the `locale` attribute to the body of the POST request:

:::: tabs
::: tab Request

**Example request**

```js
POST http://localhost:1337/restaurants
```

```json
{
  "name": "She's Cake",
  "description": "She's Cake restaurant description in French",
  "locale": "fr"
}
```

:::

::: tab Response
**Example response**

```json
{
    "id": 8,
    "name": "She's Cake",
    "description": "She's Cake restaurant description in French",
    "locale": "fr",
    "localizations": []
}
```

:::
::::

### Creating a localization for an existing entry

To create another localization for an existing localized entry, send a POST request to the appropriate URL depending on the type of content:

| Content-Type    | Request URL format                       |
|:----------------|:-----------------------------------------|
| Collection type | `POST /{content-type}/:id/localizations` |  
| Single type     | `POST /{content-type}/localizations`     |

When creating a localization for existing localized entries, the body of the POST request can only accept localized fields.

::: tip TIP

> The Content-Type should have the [`createlocalization` permission](http://localhost:8080/documentation/user-docs/latest/users-roles-permissions/configuring-administrator-roles.md#collection-and-single-types) enabled, otherwise the POST request will return a `403: Forbidden` status.

Reviewer: used a localhost:8080 address here :P

Author: Oh, nice catch! Thanks! [fixed it](https://github.com/strapi/documentation/pull/224/commits/14feba291ca9bab86b90e7df0e66d42d1e491159)

## Configuration in production environments

A `STRAPI_PLUGIN_I18N_INIT_LOCALE_CODE` [environment variable](/developer-docs/latest/setup-deployment-guides/configurations.md#environment-variables) can be configured to set the initialization locale for your environment. The value used for this variable should be a string (see [full list of available locales](https://github.com/strapi/strapi/blob/features/i18n/packages/strapi-plugin-i18n/constants/iso-locales.json)).

> This is useful when a Strapi app is deployed in production, with the i18n plugin installed and enabled on your content types for the first time. On a fresh i18n plugin installation, `'en'` is the default locale. So if the database does not contain any locale, and no `STRAPI_PLUGIN_I18N_INIT_LOCALE_CODE` is set for the environment, all the existing content saved in the database will be migrated with the locale defined as `'en'`. But if the `STRAPI_PLUGIN_I18N_INIT_LOCALE_CODE` is defined, then the content will be migrated and set up with this locale. This will prevent from manually updating the locale for every existing content entry.

Reviewer 6: instead of all the existing content I would have said that the content of the content types with i18n enabled will be auto migrated to the locale. The rest looks good to me!

Author: Thanks @alexandrebodin ! I've just [updated it](https://github.com/strapi/documentation/pull/224/commits/8001b388c89a2ef55c0166656daf16bc888c1f9b) :-)
